src/werkzeug/datastructures.py

```diff
@@ -355,6 +355,29 @@ def __setstate__(self, value):
         dict.clear(self)
         dict.update(self, value)
 
+    def __iter__(self):
+        """This looks like a no-op, but actually preserves required interop.
+
+        Without this change::
+
+            >>> d = MultiDict([('a', 'b'), ('a', 'c')])
+            >>> dict(d)  # wrong:
+            {'a': ['b', 'c']}
+            >>> {**d}  # wrong:
+            {'a': ['b', 'c']}
+
+        With this change::
+
+            >>> d = MultiDict([('a', 'b'), ('a', 'c')])
+            >>> dict(d)  # right:
+            {'a': 'b'}
+            >>> {**d}  # right:
+            {'a': 'b'}
+
+        See https://github.com/pallets/werkzeug/pull/2043.
+        """
+        return super().__iter__()
```

Comment:
Given all the other methods are `dict.xyz` I think this should be too. 
Maybe another PR to switch them all to `super`?

Reply:
There is plenty of other existing usage of `super` throughout this module (e.g. in `ImmutableDictMixin`, `UpdateDictMixin`, etc.), and I thought the `super` form makes it look even more obvious that this is a hack to work around something weird going on with the language (i.e. this code *should* be a no-op).

But I'm fine changing this to `return dict.__iter__(self)` (done in the latest revision), and letting a separate PR handle the desired conversions to `super`.